api/ruxos_posix_api/src/imp/net.rs

```diff
@@ -601,6 +621,67 @@ pub unsafe fn sys_getsockname(
     })
 }
 
+/// get socket option
+///
+/// TODO: some options not impl, just return 0, like SO_RCVBUF SO_SNDBUF
+pub fn sys_getsockopt(
+    socket_fd: c_int,
+    level: c_int,
+    optname: c_int,
+    optval: *mut c_void,
+    optlen: *mut ctypes::socklen_t,
+) -> c_int {
+    unsafe {
+        info!(
+            "sys_getsockopt <= fd: {}, level: {}, optname: {}, optlen: {}, IGNORED",
+            socket_fd,
+            level,
+            optname,
+            core::ptr::read(optlen as *mut usize)
+        );
+    }
+    syscall_body!(sys_getsockopt, {
+        return Ok(0);
+        if optval.is_null() {
+            return Err(LinuxError::EFAULT);
+        }
+        let socket = Socket::from_fd(socket_fd)?;
+        match level as u32 {
+            ctypes::SOL_SOCKET => {
+                let val = match optname {
+                    SO_ACCEPTCONN => match &*socket {
+                        Socket::Udp(_) => 0,
+                        Socket::Tcp(tcpsocket) => {
+                            if tcpsocket.lock().is_listening() {
+                                1
+                            } else {
+                                0
+                            }
+                        }
+                    },
+                    SO_TYPE => match &*socket {
+                        Socket::Udp(_) => ctypes::SOCK_DGRAM,
+                        Socket::Tcp(_) => ctypes::SOCK_STREAM,
+                    },
+                    SO_RCVLOWAT | SO_SNDLOWAT => 1,
+                    SO_BROADCAST => 1,
```

Comment:
Is there any reason not to merge `SO_RCVLOWAT | SO_SNDLOWAT => 1,` and `SO_BROADCAST => 1,`.

Reply:
I thinked it may not impl in same time in the future, but merge them is also ok.